src/ensembl/src/content/app/custom-download/components/checkbox-with-textfields/CheckboxWithTextfields.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useEffect } from 'react';
+import Checkbox from 'src/shared/checkbox/Checkbox';
+import styles from './CheckboxWithTextfields.scss';
+
+import ImageButton from 'src/shared/image-button/ImageButton';
+import { ReactComponent as AddIcon } from 'static/img/browser/zoom-in.svg';
+import PasteOrUpload from '../paste-or-upload/PasteOrUpload';
+
+export type CheckboxWithTextfieldsProps = {
+  values: string[];
+  label: string;
+  allowMultiple: boolean;
+  disabled?: boolean;
+  onChange: (values: string[]) => void;
+};
+
+const CheckboxWithTextfields = (props: CheckboxWithTextfieldsProps) => {
+  const [isChecked, setIsChecked] = useState(false);
+  const [values, setValues] = useState<(string | null)[]>(props.values);
+
+  const [shouldShowAddButton, setShowAddButton] = useState(
+    Boolean(values[values.length - 1])
+  );
+
+  useEffect(() => {
+    if (!props.values.length) {
+      setValues([null]);
+    } else {
+      setValues(props.values);
+    }
+
+    setIsChecked(props.values.length > 0);
+    setShowAddButton(Boolean(props.values[props.values.length - 1]));
+  }, [props.values]);
+
+  const handleCheckboxOnChange = (isChecked: boolean) => {
+    setIsChecked(isChecked);
+    if (!isChecked && values.length > 0) {
+      props.onChange([]);
+    }
+  };
+
+  const addEntry = () => {
+    setShowAddButton(false);
+    setValues([...values, null]);
+  };
+
+  const handleOnChange = (value: string, index: number) => {
+    const newValues: string[] = [...values] as string[];
+    newValues[index] = value;
+
+    setShowAddButton(Boolean(newValues[newValues.length - 1]));
+
+    props.onChange(newValues);
+  };
+
+  const handleOnRemove = (index: number) => {
+    const newValues: string[] = [...values] as string[];
+
+    newValues.splice(index, 1);
+    setShowAddButton(Boolean(newValues[newValues.length - 1]));
+
+    props.onChange(newValues);
+  };
+
+  const valuesWithoutFirst: string[] = [...values] as string[];
+  valuesWithoutFirst.shift();
```

Comment:
`const valuesWithoutFirst = values.slice(1);`
but why do you need them?

Reply:
That's not needed anymore. The logic was a bit different initially and I missed to remove this when I changed it.